text/0003-font-types.md

```diff
@@ -0,0 +1,408 @@
+- 2022-01-17
+- status: proposal
+- PR: https://github.com/googlefonts/oxidize/pull/3
+
+## A foundation for building font tooling in Rust
+
+
+# Overview
+
+We propose a Rust library (crate) for zero-allocation access and direct manipulation
+of font tables and types, leveraging Rust’s procedural macro system. This
+library is intended to be an opinion-free transparent representation of the types
+in the specification, to serve as a foundational component for anything that
+needs to read or write font files. An additional goal is to provide a common set
+of types to represent common data formats (tags, glyph identifiers, etc) that
+can be used throughout the broader ecosystem.
+
+
+# Motivation
+
+This is intended to answer some of the questions outlined in this repository's
+README. In particular, we aim to determine whether the following goals are feasible:
+
+- using macros to describe the shape of tables
+-copy generating both zero-allocation read-only table definitions as well as owned,
+  mutable definitions (for compilation)
+- suitable for prototyping & extensibility, i.e. it should be easy to add a new
```

Comment:
This part isn't a core goal, that is in the original problem statement it would be fine to require hacking on the core (akin hacking on otTables in FontTools).

It IS a worthy thing to achieve but maybe it should be called out as a goal somewhat beyond the core goal.

Reply:
I *think* we should just get this for free as a consequence of the design, but can definitely add an 'additional goals' section.